test/alphabet/alphabet_test.cpp

```diff
@@ -154,8 +154,9 @@ TYPED_TEST(alphabet, swap)
 
 TYPED_TEST(alphabet, assign_char)
 {
+    using char_t = typename TypeParam::char_type;
     TypeParam t0;
-    for (char i = std::numeric_limits<char>::min(); i < std::numeric_limits<char>::max(); ++i)
+    for (char_t i = std::numeric_limits<char_t>::min(); i < std::numeric_limits<char_t>::max(); ++i)
```

Comment:
@mariehoffmann This test triggers a warning for the illumina 1.8 quality alphabet. `assign_char` is not defined for all possible chars (only for `!` - `J`). Is there an unknown value in the alphabet that we can set the value to in case an unknown/invalid character is passed to `assign_char`?

I think this might be a problem if the user is building in release mode since assertions are only checked in debug mode.